docs/pages/admin-guides/access-controls/idps/saml-microsoft-entra-external-id.mdx

```diff
@@ -0,0 +1,235 @@
+
+![Microsoft Entra External ID link subscription](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-link-subscription.png)
+
+In the Azure portal, under **Microsoft Entra External ID** page, link an active subscription to External ID. 
+Make sure to select the same subscription that you selected while creating the resource group in Step 4. 
+
+Once the subscription is linked, user `alice@example.com` will be able to access VM resources available
+under `teleport-dev-vm-pool` resource group.
+
+## Azure CLI access
+
+Users can authenticate Azure CLI with Teleport by providing a tenant ID to `az login` command.
+```bash
+az login --tenant <tenant_id>
+```
+
+The command will open a web browser window that redirects the user to authenticate with Teleport SAML IdP. 
+Upon successful authentication, Microsoft Entra ID responds with credentials for the CLI.
+
+## User access audit logs
+
+User access logs can be found in both Teleport Web UI and in Azure portal.
+
+In Teleport, each user authentication with SAML IdP is logged and the event can be found in the audit log.
+```json
+{
+  "cluster_name": "mytenant.teleport.sh",
+  "code": "TSI000I",
+  "ei": 0,
+  "event": "saml.idp.auth",
+  "service_provider_entity_id": "https://login.microsoftonline.com/<tenant_id>>/",
+  "sid": "",
+  "success": true,
+  "time": "2025-04-15T14:43:40.057Z",
+  "uid": "d6fa48a4-8bfd-420c-992c-24bba11fb30a",
+  "user": "alice@example.com"
+}
+```
+
+The same login event can also be found in Azure portal under **User > Sign-in logs** page.
+
+## FAQ
+
+### How to automate external user invitation
+
+Users can be bulk invited using the [Microsoft Graph API](https://learn.microsoft.com/en-us/graph/api/invitation-post?view=graph-rest-1.0&tabs=powershell#example-1-invite-a-guest-user).
+
+### Associating multiple domain names
+
+Multiple domain can be federated as long as DNS of those domain name is updated to let the Microsoft 
+Entra ID knows that each such domain is associated with Teleport SAML IdP. 
+
+First, open domain configuration menu for the selected identity provider
+![Entra ID update IdP domain](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-update-domain.png)
+
+Then add a new domain name.
+![Entra ID associate new domain with the IdP](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-add-new-domain.png)
+
+Note that the rules of DNS update apply here as well. 
+
+## Troubleshooting
+
+### Issues related to identity provider configuration
+
+Refer to official [Microsoft Entra External ID SAML federation](https://learn.microsoft.com/en-us/entra/external-id/direct-federation) docs for general configuration.
+
+### Issues related to user invitation and user type conversion
+For issues related to user invitation, or internal to external user type conversion, please refer to [Microsoft Entra B2B collaboration](https://learn.microsoft.com/en-us/entra/external-id/troubleshoot) troubleshooting docs.
```

Comment:
```suggestion
For issues related to user invitation, or internal to external user type conversion, refer to the [Microsoft Entra B2B collaboration](https://learn.microsoft.com/en-us/entra/external-id/troubleshoot) troubleshooting docs.
```
This needs an article. Also removing "please" since it's not needed in documentation.